src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -0,0 +1,213 @@
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	type wrappedResponse struct {
+		Error    string                       `json:"error"`
+		Metadata apiv1.WorkerNodeInfoResponse `json:"metadata"`
+	}
+
+	requestBody, err := json.Marshal(apiv1.WorkerNodeInfoRequest{Hostname: s.Name()})
+	if err != nil {
+		return fmt.Errorf("failed to prepare worker info request: %w", err)
+	}
+
+	httpRequest, err := http.NewRequest("POST", fmt.Sprintf("https://%s/1.0/k8sd/worker/info", token.JoinAddresses[0]), bytes.NewBuffer(requestBody))
+	if err != nil {
+		return fmt.Errorf("failed to prepare HTTP request: %w", err)
+	}
+	httpRequest.Header.Add("k8sd-token", token.Token)
+
+	httpResponse, err := httpClient.Do(httpRequest)
+	if err != nil {
+		return fmt.Errorf("failed to POST %s: %w", httpRequest.URL.String(), err)
+	}
+	defer httpResponse.Body.Close()
+	var wrappedResp wrappedResponse
+	if err := json.NewDecoder(httpResponse.Body).Decode(&wrappedResp); err != nil {
+		return fmt.Errorf("failed to parse HTTP response: %w", err)
+	}
+	if httpResponse.StatusCode != 200 {
+		return fmt.Errorf("HTTP request for worker node info failed: %s", wrappedResp.Error)
+	}
+	response := wrappedResp.Metadata
+
+	snap := snap.SnapFromContext(s.Context)
+	if err := setup.InitFolders(snap.DataPath("args")); err != nil {
+		return fmt.Errorf("failed to setup folders: %w", err)
+	}
+	if err := setup.InitContainerd(snap.Path("k8s/config/containerd/config.toml"), snap.Path("opt/cni/bin/")); err != nil {
+		return fmt.Errorf("failed to configure containerd: %w", err)
+	}
+	if err := setup.InitContainerdArgs(snap, nil, nil); err != nil {
+		return fmt.Errorf("failed to configure containerd arguments: %w", err)
+	}
+	if err := setup.WriteCA(snap, response.CA); err != nil {
+		return fmt.Errorf("failed to write CA certificate: %w", err)
+	}
+
+	kubeletArgs := map[string]string{
+		"--hostname-override": s.Name(),
+		"--cluster-dns":       response.ClusterDNS,
+		"--cluster-domain":    response.ClusterDomain,
+		"--cloud-provider":    response.CloudProvider,
+	}
+	if err := setup.InitKubeletArgs(snap, kubeletArgs, nil); err != nil {
+		return fmt.Errorf("failed to configure kubelet: %w", err)
+	}
+	if err := setup.RenderKubeletKubeconfig(snap, response.KubeletToken, response.CA); err != nil {
+		return fmt.Errorf("failed to render kubelet kubeconfig: %w", err)
+	}
+
+	proxyArgs := map[string]string{
+		"--hostname-override": s.Name(),
+		"--cluster-cidr":      response.ClusterCIDR,
+	}
+	if err := setup.InitKubeProxyArgs(snap, proxyArgs, nil); err != nil {
+		return fmt.Errorf("failed to configure kube-proxy: %w", err)
+	}
+	if err := setup.RenderKubeProxyKubeconfig(snap, response.KubeProxyToken, response.CA); err != nil {
+		return fmt.Errorf("failed to render kube-proxy kubeconfig: %w", err)
+	}
+
+	if err := setup.InitAPIServerProxy(snap, response.APIServers); err != nil {
+		return fmt.Errorf("failed to configure k8s-apiserver-proxy: %w", err)
+	}
+
+	// TODO: mark node as worker
```

Comment:
the `k8sd` database will be there because we have initialized a microcluster app locally. `k8s-dqlite` might also not be in use, (e.g. if using an external etcd).

It's best to declare _intent_ rather than rely on figuring this out from side-effects.

Reply:
fair.